### DISCUSSION TITLE
Add an option to generate component deterministically

### DISCUSSION THREAD

**Discussion by zeid0ne** [quasarframework/quasar#17377] [Ideas / Proposals]
Let's take QExpansionItem for example.

When generating corresponding div, Quasar add some attributes to some div that are random, for example "id" or "aria-controls".

I am using Vitest, and i am generating a snapshot of a component using a QExpansionItem.

So, every time i run the snapshot, i have a different result, which break the goal of the snapshot.

A workaround is to stub the QExpansionItem, but thats not perfect cause my snapshot now cant detect "normal" changes about the QExpansionItem.

Is it possible to add an option to Quasar, in order to generate component deterministically please ?

Or even better is it possible to generate component determinstically by default ?

Thanks for all

**Comment by ivanjaros** [comment]
i think the random stuff is added only if you use scoped css?

**Reply by zeid0ne** [reply]
yes i use scoped css
